config/clock.rb

```diff
@@ -22,6 +22,7 @@ class Clock
   every(1.hour, 'SendCourseFullNotifications', at: '**:45') { SendCourseFullNotificationsWorker.perform_async }
 
   every(1.hour, 'RejectAwaitingReferencesCourseChoices', at: '**:50') { RejectAwaitingReferencesCourseChoicesWorker.perform_async }
+  every(1.day, 'CarryOverUnsubmittedApplications', at: '00:01', if: ->(t) { t.to_date == EndOfCycleTimetable.date(:next_cycle_opens) }) { CarryOverUnsubmittedApplicationsWorker.perform_async }
```

Comment:
Wondering if we should do the same for the `RejectAwaitingReferencesCourseChoicesWorker` above?

Reply:
Yes it's running redundantly most of the time. Does it need to happen on the same day? When the cycle opens or earlier than that?

Reply:
I think it should the `RejectAwaitingReferencesCourseChoicesWorker` should run after apply 2 closes (19 Sept). Slightly more complicated than I thought so going to raise a separate PR and get this one merged.

https://github.com/DFE-Digital/apply-for-teacher-training/compare/1959-change-reject-awaiting-refs-to-oneoff-job?expand=1